Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-roleserversideviewservice', function (Y) {
+    "use strict";
+    /**
+     * Provides the role server side view service class
+     *
+     * @method ez-roleserversideviewservice
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The Role Server Side View Service class.
+     *
+     * @namespace eZ
+     * @class RoleServerSideViewService
+     * @constructor
+     * @extends eZ.ServerSideViewService
+     */
+    Y.eZ.RoleServerSideViewService = Y.Base.create('roleServerSideViewService', Y.eZ.ServerSideViewService, [], {
+        initializer: function () {
+            this.on('*:contentDiscover', function (e) {
+                e.config.contentDiscoveredHandler = Y.bind(this._assignRole, this);
+            });
+        },
+
+        /**
+         * Assign the role to the contents after the user has chosen them in
+         * the universal discovery widget
+         *
+         * @method _assignRole
+         * @protected
+         * @param {EventFacade} e
+         */
+        _assignRole: function (e) {
+            var data = e.target.get('data'),
+                userService = this.get('capi').getUserService(),
+                discoveryService = this.get('capi').getDiscoveryService(),
+                service = this;
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            userService.loadRole(data.roleId, function (error, role) {
```

Comment:
`userService.loadRole` returns response from REST, so the Role is embed inside the response. This line should rather looks like:
```javascript
userService.loadRole(data.roleId, function (error, response) {
  var tasks =  new Y.Parallel(),
      end = tasks.add( /* function */),
      role;

  if (error) {
    /* error handling */
  }

  role = response.document.Role;
  /* iterate through e.selection */
        
});
```